src/OpenTelemetry.AutoInstrumentation.Loader/AppConfigUpdater.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#if NETFRAMEWORK
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+
+namespace OpenTelemetry.AutoInstrumentation.Loader;
+
+#pragma warning disable RS0016
+#pragma warning restore RS0016
+
+/// <summary>
+/// Handles update of config files for non-default AppDomain
+/// </summary>
+public static class AppConfigUpdater
```

Comment:
If we are calling this via reflection, does this still need to be part of our public API?

Reply:
It does not need to be public, but making it private will make profiler-generated code that resolves it harder, as it will need use GetType/GetMethod overloads that acceps `BindingFlags` and load const with `BindingFlags` allowing private members.
On other hand, the type is defined in `OpenTelemetry.AutoInstrumentation.Loader`, which is not distributed separetely in .Net Framework (available only as profiler resource). Type is not defined in .Net version of assembly, so it still hard to access by any other means than reflection.
I can make it private if it's still desired.

Reply:
Since it is not that simple, I am fine with keeping it public. That will help keep the injected code easier to read.

Reply:
Can we work on this as a follow up?
Especially making method private/internal. I would like to prevent manual calls to this method.

Reply:
Actually we can. I missed, that we don't need any binder for resolving internal type. So, making AppConfigUpdater interanl (still leaving method inside public).